nameko/containers.py

```diff
@@ -127,9 +127,25 @@ def __init__(self, service_cls, worker_ctx_cls, config):
         self.max_workers = (
             config.get(MAX_WORKERS_CONFIG_KEY) or DEFAULT_MAX_WORKERS)
 
-        self.dependencies = DependencySet()
-        for dep in prepare_dependencies(self):
-            self.dependencies.add(dep)
+        self.entrypoints = SpawningSet()
+        self.dependencies = SpawningSet()
+        self.subextensions = SpawningSet()
```

Comment:
I've done away with the ``ExtensionSet`` and we now deal with the three categories explicitly. I'm not super keen on ``subextensions`` as a name but that is what they are.